celery/backends/elasticsearch.py

```diff
@@ -88,6 +89,7 @@ def set(self, key, value):
             self._index(key, data, refresh=True)
 
     def _index(self, id, body, **kwargs):
+        body = dict((string(k), v) for k, v in body.items())
```

Comment:
This line can be simplified by using a dictionary comprehension (Celery 4 requires Python 2.7 and above). Also, you can avoid creating a new list in Python 2 by using the compatibility function `items` from `five`:

```
from celery.five import items

body = {string(key): value for key, value in items(body)}
```

What do you think?